File: src/0667/S0667_v1.py
class Solution:
    def constructArray(self, n: int, k: int) -> List[int]:
        return [res for res in self.gen(n, k)]

    def gen(self, n, k):
        yield 1
        res = 1
        for i in range(k, 0, -1):
            if k%2 == 0:
                res = res + i if i%2 == 0 else res - i
            else:
                res = res + i if i%2 == 1 else res - i
            yield res

        for j in range(k+2, n+1):
            yield j

File: src/0667/S0667_v1.rs
impl Solution {
    pub fn construct_array(n: i32, k: i32) -> Vec<i32> {        
        (1..n+1)
            .map(|x| Solution::gen(x, k))
            .collect::<Vec<i32>>()
    }

    pub fn gen(r: i32, k: i32) -> i32 {
        if r == 1 {
            return 1;
        }
        else if r <= k+1 {
            if r%2 == 0 {
                return 2 + k - (r / 2);
            } else {
                return (r + 1) / 2;
            }
        } else {
            return r;
        }
    }
}
